relayer/crates/starknet-test-components/src/impls/bootstrap_madara/start_madara.rs

```diff
@@ -5,50 +5,71 @@ use hermes_cosmos_test_components::bootstrap::traits::chain::start_chain::{
 use hermes_cosmos_test_components::bootstrap::traits::fields::chain_command_path::HasChainCommandPath;
 use hermes_cosmos_test_components::bootstrap::traits::types::chain_node_config::HasChainNodeConfigType;
 use hermes_cosmos_test_components::bootstrap::traits::types::genesis_config::HasChainGenesisConfigType;
-use hermes_runtime_components::traits::fs::file_path::HasFilePathType;
+use hermes_runtime_components::traits::fs::create_dir::CanCreateDir;
 use hermes_runtime_components::traits::os::child_process::CanStartChildProcess;
 use hermes_runtime_components::traits::runtime::HasRuntime;
 
-use crate::types::genesis_config::StarknetGenesisConfig;
 use crate::types::node_config::StarknetNodeConfig;
 
 #[cgp_new_provider(ChainFullNodeStarterComponent)]
-impl<Bootstrap, Runtime> ChainFullNodeStarter<Bootstrap> for StartMadaraDevnet
+impl<Bootstrap, Runtime> ChainFullNodeStarter<Bootstrap> for StartMadaraSequencer
 where
     Bootstrap: HasRuntime<Runtime = Runtime>
         + HasChainNodeConfigType<ChainNodeConfig = StarknetNodeConfig>
-        + HasChainGenesisConfigType<ChainGenesisConfig = StarknetGenesisConfig>
+        + HasChainGenesisConfigType
         + HasChainCommandPath
         + CanRaiseAsyncError<Runtime::Error>,
-    Runtime: CanStartChildProcess + HasFilePathType,
+    Runtime: CanStartChildProcess + CanCreateDir,
 {
-    async fn start_chain_full_node(
+    async fn start_chain_full_nodes(
         bootstrap: &Bootstrap,
         chain_home_dir: &Runtime::FilePath,
         chain_node_config: &StarknetNodeConfig,
-        chain_genesis_config: &StarknetGenesisConfig,
-    ) -> Result<Runtime::ChildProcess, Bootstrap::Error> {
+        chain_genesis_config: &Bootstrap::ChainGenesisConfig,
+    ) -> Result<Vec<Runtime::ChildProcess>, Bootstrap::Error> {
         let chain_command = bootstrap.chain_command_path();
 
+        let madara_home =
+            Runtime::join_file_path(chain_home_dir, &Runtime::file_path_from_string("madara"));
+
+        bootstrap
+            .runtime()
+            .create_dir(&madara_home)
+            .await
+            .map_err(Bootstrap::raise_error)?;
+
+        let rpc_port = chain_node_config.rpc_port;
+
+        let gateway_port = rpc_port + 1;
+
         let args = [
-            "--devnet",
-            "--rpc-port",
-            &chain_node_config.rpc_port.to_string(),
             "--base-path",
-            &Runtime::file_path_to_string(chain_home_dir),
+            &Runtime::file_path_to_string(&madara_home),
+            "--rpc-port",
+            &rpc_port.to_string(),
+            "--gateway-port",
+            &gateway_port.to_string(),
             "--chain-config-override",
-            "block_time=1s,pending_block_update_time=1s",
+            "block_time=1s,pending_block_update_time=1s,chain_id=SN_SEPOLIA",
```

Comment:
you pass `chain_id` to be `SN_SEPOLIA` here ..

Reply:
We also need this here, because Pathfinder would only work if it is connected to the Sepolia testnet. The main thing we need to be careful is that the signed transactions can also work on the testnet. But it is fine here because we are using test wallets generated by Madara.